src/sensors/mpu9250sensor.cpp

```diff
@@ -214,6 +217,44 @@ void MPU9250Sensor::getMPUScaled()
 
 void MPU9250Sensor::startCalibration(int calibrationType) {
     ledManager.on();
+#if not (defined(_MAHONY_H_) || defined(_MADGWICK_H_))
+    // with DMP, we just need mag data
+    constexpr int calibrationSamples = 300;
+    DeviceConfig *config = getConfigPtr();
+    // Blink calibrating led before user should rotate the sensor
+    m_Logger.info("Gently rotate the device while it's gathering magnetometer data");
+    ledManager.pattern(15, 300, 3000/310);
+    float *calibrationDataMag = (float*)malloc(calibrationSamples * 3 * sizeof(float));
+    for (int i = 0; i < calibrationSamples; i++)
+    {
+        ledManager.on();
+        int16_t ax,ay,az,gx,gy,gz,mx,my,mz;
+        imu.getMagnetometer(&mx, &my, &mz);
+        calibrationDataMag[i * 3 + 0] = my;
+        calibrationDataMag[i * 3 + 1] = mx;
+        calibrationDataMag[i * 3 + 2] = -mz;
+        Network::sendRawCalibrationData(calibrationDataMag, CALIBRATION_TYPE_EXTERNAL_MAG, 0);
+        ledManager.off();
+        delay(250);
+    }
+    m_Logger.debug("Calculating calibration data...");
+
+    float M_BAinv[4][3];
+    CalculateCalibration(calibrationDataMag, calibrationSamples, M_BAinv);
+    free(calibrationDataMag);
+
+    m_Logger.debug("[INFO] Magnetometer calibration matrix:");
+    m_Logger.debug("{");
+    for (int i = 0; i < 3; i++) {
```

Comment:
Please format, either you put `{` on the next line or on the current line, not both.

Reply:
I didn't code that. that is original code. but I'll fix it.